RNSentry.podspec

```diff
@@ -14,7 +14,7 @@ is_new_arch_enabled = ENV["RCT_NEW_ARCH_ENABLED"] == "1"
 is_using_hermes = (ENV['USE_HERMES'] == nil && is_hermes_default) || ENV['USE_HERMES'] == '1'
 new_arch_enabled_flag = (is_new_arch_enabled ? folly_compiler_flags + " -DRCT_NEW_ARCH_ENABLED" : "")
 sentry_profiling_supported_flag = (is_profiling_supported ? " -DSENTRY_PROFILING_SUPPORTED=1" : "")
-other_cflags = "$(inherited)" + new_arch_enabled_flag + sentry_profiling_supported_flag
+other_cflags = "$(inherited) -fcxx-modules -fmodules" + new_arch_enabled_flag + sentry_profiling_supported_flag
```

Comment:
We can't add these flag, because some of the React Native source code is not compatible with it. 

Anywhere where you need to use the Sentry module isolate it in `.m` Obj-C file and then import it in `RNSentry.mm`.

Reply:
We need to implement `SentryReplayBreadcrumbConverter` which is a swift protocol. How can we use that in RN? 

I suppose @brustolin hase more insights into this.

Reply:
As @krystofwoldrich explained. You need to create a new Objc class that will be responsible for dealing with Sentry module, and call this new object from the RNSentry.mm

We need to stop adding behaviour to `RNSentry.mm` and use it only as a bridge between RN and Native code.

We cant `@import Sentry;` from any header that will be used from `RNSentry.mm` but we dont need to.

Reply:
How does this solve the issue of not having access to modules? The breadcrumb converter is already a separate class RNSentryBreadcrumbConverter.h/.m

Reply:
This is what the build fails with when `-fcxx-modules -fmodules` is removed:

![image](https://github.com/getsentry/sentry-react-native/assets/6349682/0cb9849d-2dce-4606-b0be-aa2247dd8c17)

Reply:
> We cant @import Sentry; from any header that will be used from RNSentry.mm but we dont need to.

Missed this, let me give it a try

Reply:
that worked, thx @brustolin 

https://github.com/getsentry/sentry-react-native/pull/3846/commits/3d76fae9fdcbdfc94a18cd9dd6a481c60f3459fa